libs/utils/utils.js

```diff
@@ -438,31 +438,35 @@ export function decorateAutoBlock(a) {
         a.dataset.modalPath = url.pathname;
         a.dataset.modalHash = url.hash;
         a.href = url.hash;
-        a.className = 'modal link-block';
-        return true;
+        a.classList.add('modal', 'link-block');
+      } else {
+        a.classList.add(key, 'link-block');
       }
-      a.className = `${key} link-block`;
       return true;
     }
     return false;
   });
 }
 
-export function decorateLinks(el) {
+export async function decorateLinks(el) {
   const anchors = el.getElementsByTagName('a');
-  return [...anchors].reduce((rdx, a) => {
+  const decorateModule = import('./decorate.js');
+  const links = [...anchors].reduce((rdx, a) => {
     a.href = localizeLink(a.href);
     decorateSVG(a);
     if (a.href.includes('#_blank')) {
       a.setAttribute('target', '_blank');
       a.href = a.href.replace('#_blank', '');
     }
     const autoBLock = decorateAutoBlock(a);
+    decorateModule.then((mod) => { mod.decorateLinkToButton(a); });
     if (autoBLock) {
       rdx.push(a);
     }
     return rdx;
   }, []);
+  await decorateModule;
+  return links;
```

Comment:
The above reads a bit weird to me.
```js
export async function decorateLinks(el) {
  const anchors = el.getElementsByTagName('a');
  const { decorateLinkToButton } = await import('./decorate.js');
  return [...anchors].reduce((rdx, a) => {
    a.href = localizeLink(a.href);
    decorateSVG(a);
    if (a.href.includes('#_blank')) {
      a.setAttribute('target', '_blank');
      a.href = a.href.replace('#_blank', '');
    }
    const autoBlock = decorateAutoBlock(a);
    decorateLinkToButton(a);
    if (autoBlock) rdx.push(a);
    return rdx;
  }, []);
}
```
Does this not work?

Reply:
It does, but it causes weird timing problems with quite a few unit tests, especially in the gnav.

Reply:
Maybe @chrischrischris and I can look at it today.

Gnav's decorateLinks use needs to die anyway... it's causing performance problems.

Reply:
That would be awesome ! You're taking a weight off my chest :)